### DISCUSSION TITLE
Storybook next-intl addon

### DISCUSSION THREAD

**Discussion by stevensacks** [amannn/next-intl#1214] [Show and tell]
Hello  @amannn,

I've published a Storybook addon for next-intl. It's based on my storybook-i18n addon, and like my other i18n addons, it's easy to integrate, and full instructions are in the README.

https://github.com/stevensacks/storybook-next-intl

Please check it out, and once you're happy with it, it would be great if you'd consider adding it to your docs regarding Storybook integration.

Best,
Steven

**Comment by amannn** [comment]
Oh that’s really cool, thanks for sharing! Would be more than happy to test it and give it a shout-out in the docs! I’m currently on vacation for a few weeks, I’ll be in touch again afterwards!

**Comment by amannn** [comment]
@stevensacks I finally had a chance to have a closer look, `storybook-next-intl` looks fantastic! 👏

I've set up a PR to add a docs page about Storybook that references your addon: https://github.com/amannn/next-intl/pull/1266. Mind having a look at the proposed page if this looks good to you?

While trying out the setup, I noticed a few things that you could potentially address with the addon:
1. In the README there's a discrepancy of whether the file should be called `nextIntl.ts` or `next-intl.ts`. Even if it's up to the user, I'd suggest sticking with one variation to avoid confusion.
2. I'd consider renaming the config option `messages` to `messagesByLocale`. `messages` is a term we use in `next-intl` to refer to the messages for a given locale, e.g. the ones that are assigned to `NextIntlClientProvider`. Choosing a different name for your config option might make it clearer that this is something different.
3. In the docs, the message import path points to `'../src/messages/en'` (etc.), I'd suggest changing it to `'../messages/en.json'`. This would be aligned with the defaults we suggest to users (`./messages` in root folder, `.json` extension). Again, it's really up to the user, but for those who use the default config, this might make copy-pasting the instructions more straightforward.
4. As far as I understand, it's currently not possible to provide other configuration to `NextIntlClientProvider` (e.g. `formats`). Maybe you could allow specifying those in `next-intl.ts`.

That's all I noticed, hope this is helpful to you!

**Reply by stevensacks** [reply]
Thanks for the feedback. I’ll update it shortly and let you know when it’s ready.

**Comment by stevensacks** [comment]
@amannn https://github.com/stevensacks/storybook-next-intl/releases/tag/v1.1.4

I added support for all of the optional provider props and updated the documentation accordingly.

Thanks!

**Reply by amannn** [reply]
Awesome! 🙌 

The Storybook page is now [live](https://next-intl-docs.vercel.app/docs/workflows/storybook) and I've [announced it on X](https://x.com/jamannnnnn/status/1826596977465061847).
